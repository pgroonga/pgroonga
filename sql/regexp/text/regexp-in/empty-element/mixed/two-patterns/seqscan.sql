CREATE TABLE memos (
  content text
);

INSERT INTO memos VALUES ('PostgreSQL is a RDBMS');
INSERT INTO memos VALUES ('RDBMS is abbreviation of Relational Database Management System');
INSERT INTO memos VALUES ('Groonga is fast full text search engine');
INSERT INTO memos VALUES ('PGroonga is a PostgreSQL extension that uses Groonga');

SET enable_seqscan = on;
SET enable_indexscan = off;
SET enable_bitmapscan = off;

EXPLAIN (COSTS OFF)
SELECT *
  FROM memos
 WHERE content &~| ARRAY['', 'rdbms\z', '\Agroonga'];

SELECT *
  FROM memos
 WHERE content &~| ARRAY['', 'rdbms\z', '\Agroonga'];

DROP TABLE memos;
