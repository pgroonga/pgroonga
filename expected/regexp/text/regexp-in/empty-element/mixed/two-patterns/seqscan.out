CREATE TABLE memos (
  content text
);
INSERT INTO memos VALUES ('PostgreSQL is a RDBMS');
INSERT INTO memos VALUES ('RDBMS is abbreviation of Relational Database Management System');
INSERT INTO memos VALUES ('Groonga is fast full text search engine');
INSERT INTO memos VALUES ('PGroonga is a PostgreSQL extension that uses Groonga');
SET enable_seqscan = on;
SET enable_indexscan = off;
SET enable_bitmapscan = off;
EXPLAIN (COSTS OFF)
SELECT *
  FROM memos
 WHERE content &~| ARRAY['', 'rdbms\z', '\Agroonga'];
                           QUERY PLAN                           
----------------------------------------------------------------
 Seq Scan on memos
   Filter: (content &~| '{"","rdbms\\z","\\Agroonga"}'::text[])
(2 rows)

SELECT *
  FROM memos
 WHERE content &~| ARRAY['', 'rdbms\z', '\Agroonga'];
                 content                 
-----------------------------------------
 PostgreSQL is a RDBMS
 Groonga is fast full text search engine
(2 rows)

DROP TABLE memos;
